qiskit_research/utils/gates.py

```diff
@@ -136,6 +136,41 @@ def __array__(self, dtype=None):
         return np.array([[0, -1j], [1j, 0]], dtype=dtype)
 
 
+class PiPhiGate(Gate):
+    r"""The single-qubit X gate (:math:`\sigma_x`), implemented
+    with an angle :math:`\\phi` relative to the X-axis.
+    """
+
+    def __init__(self, phi: ParameterValueType, label: Optional[str] = None):
+        """Create new PiPhi gate."""
+        super().__init__("\\pi_{\\phi}", 1, [phi], label=label)
+        self.phi = phi
+
+    def _define(self):
+        q = QuantumRegister(1, "q")
+        qc = QuantumCircuit(q, name=self.name)
+        # rules = [(U3Gate(pi, pi / 2, pi / 2), [q[0]], [])] # FIX
+        rules = [
+            (RZGate(self.phi), [q[0], []]),
+            (XGate(), [q[0]], []),
+            (RZGate(-self.phi), [q[0]], []),
+        ]  # maybe this is right
```

Comment:
what's the meaning of this comment?

Reply:
Checked to make sure correct sign.. it is (and removed comment).